### PR TITLE
[Merged by Bors] - feat(LinearAlgebra.Span): add apply_mem_span_image_iff_mem_span

### PR DESCRIPTION
**Pull request by xroblot**
Add ~~the other direction and~~ iff version of [submodule.apply_mem_span_image_of_mem_span](https://leanprover-community.github.io/mathlib_docs/linear_algebra/span.html#submodule.apply_mem_span_image_of_mem_span), that is:
```lean
theorem apply_mem_span_image_iff_mem_span [RingHomSurjective σ₁₂] {f : M →ₛₗ[σ₁₂] M₂} {x : M} {s : Set M} 
    (hf : Function.Injective f) :  f x ∈ Submodule.span R₂ (f '' s) ↔ x ∈ Submodule.span R s
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
Please remember to add the label `awaiting-review` when this is ready for review. (Alternatively PRs that refactor files that were ported from mathlib3 should be labelled `after-port`.)

File: Mathlib/LinearAlgebra/Span.lean [hunk 1]
```diff
@@ -583,6 +583,17 @@ theorem apply_mem_span_image_of_mem_span [RingHomSurjective σ₁₂] (f : M →
   exact Submodule.mem_map_of_mem h
 #align submodule.apply_mem_span_image_of_mem_span Submodule.apply_mem_span_image_of_mem_span
 
+theorem mem_span_of_apply_mem_span_image [RingHomSurjective σ₁₂] {f : M →ₛₗ[σ₁₂] M₂} {x : M}
+    {s : Set M} (hf : Function.Injective f) (h : f x ∈ Submodule.span R₂ (f '' s)) :
+    x ∈ Submodule.span R s := by
+  rwa [← Submodule.mem_comap, ← Submodule.map_span, Submodule.comap_map_eq_of_injective hf] at h
```
Comment (ADedecker) [inline Mathlib/LinearAlgebra/Span.lean:589]: Doesn't this work direclty for the `iff` version? If it does, I would say we can keep only the `iff` version if that works for you.

**Review by ADedecker — APPROVED** [review]

File: Mathlib/LinearAlgebra/Span.lean [hunk 2]
```diff
@@ -597,6 +597,13 @@ theorem apply_mem_span_image_of_mem_span [RingHomSurjective σ₁₂] (f : M →
   exact Submodule.mem_map_of_mem h
 #align submodule.apply_mem_span_image_of_mem_span Submodule.apply_mem_span_image_of_mem_span
 
+theorem apply_mem_span_image_iff_mem_span [RingHomSurjective σ₁₂] {f : M →ₛₗ[σ₁₂] M₂} {x : M}
+    {s : Set M} (hf : Function.Injective f) :
+    f x ∈ Submodule.span R₂ (f '' s) ↔ x ∈ Submodule.span R s :=
+  ⟨fun h => by
+    rwa [← Submodule.mem_comap, ← Submodule.map_span, Submodule.comap_map_eq_of_injective hf] at h,
+    fun h => Submodule.apply_mem_span_image_of_mem_span _ h⟩
```
Comment (ADedecker) [inline Mathlib/LinearAlgebra/Span.lean:605]: Sorry I wasn't clear, I meant that the `rw` proof already worked for the equivalence:
```suggestion
    f x ∈ Submodule.span R₂ (f '' s) ↔ x ∈ Submodule.span R s := by
  rw [← Submodule.mem_comap, ← Submodule.map_span, Submodule.comap_map_eq_of_injective hf]
```

Comment (xroblot) [reply Mathlib/LinearAlgebra/Span.lean]: Oh, right. I missed that. Thanks!

**Comment by ADedecker** [comment]
Thanks!
maintainer merge

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors r+

**Pull request closed without merging**